providers/flagd/pkg/provider.go

```diff
@@ -271,11 +429,81 @@ func (p *Provider) ObjectEvaluation(
 		}
 	}
 
-	return of.InterfaceResolutionDetail{
+	resDetail := of.InterfaceResolutionDetail{
 		Value: res.Value,
 		ProviderResolutionDetail: of.ProviderResolutionDetail{
 			Reason:  of.Reason(res.Reason),
 			Variant: res.Variant,
 		},
 	}
+
+	if p.isCacheAvailable() && res.Reason == flagdModels.StaticReason {
+		p.interfaceCache.Add(flagKey, resDetail)
+	}
+
+	return resDetail
+}
+
+func (p *Provider) isCacheAvailable() bool {
+	return p.cacheEnabled && p.Service.IsEventStreamAlive()
+}
+
+func (p *Provider) handleEvents(ctx context.Context) error {
+	eventChan := make(chan *schemaV1.EventStreamResponse)
+	errChan := make(chan error)
+
+	go p.Service.EventStream(ctx, eventChan, errChan)
+
+	for {
+		select {
+		case event, ok := <-eventChan:
+			if !ok {
+				if p.cacheEnabled { // disable cache
+					WithoutCache()(p)
+				}
+				return nil
+			}
+
+			switch event.Type {
+			case string(flagdService.ConfigurationChange):
+				if err := p.handleConfigurationChangeEvent(ctx, event); err != nil {
+					log.Errorf("handle configuration change event: %v", err)
+				}
+			}
+		case err := <-errChan:
+			if p.cacheEnabled { // disable cache
+				WithoutCache()(p)
+			}
+			return err
+		}
+	}
+}
+
+func (p *Provider) handleConfigurationChangeEvent(ctx context.Context, event *schemaV1.EventStreamResponse) error {
+	if !p.cacheEnabled {
+		return nil
+	}
+
+	if event.Data == nil {
+		return errors.New("no data in event")
+	}
+
+	flagKeyVal, ok := event.Data.AsMap()["flagKey"]
+	if !ok {
+		return errors.New("no flagKey field in event data")
+	}
+
+	flagKey, ok := flagKeyVal.(string)
+	if !ok {
+		return errors.New("flagKey is not a string")
+	}
+
+	// TODO: consider sending the flag type in the configuration change event
```

Comment:
potential optimisation

Reply:
This is impacted by https://github.com/open-feature/flagd/issues/238